nifi-nar-bundles/nifi-aws-bundle/nifi-aws-processors/src/main/java/org/apache/nifi/processors/aws/AbstractAWSProcessor.java

```diff
@@ -161,6 +180,12 @@ private static AllowableValue createAllowableValue(final Regions regions) {
             problems.add(new ValidationResult.Builder().input("Access Key").valid(false).explanation("Cannot set both Credentials File and Secret Key/Access Key").build());
         }
 
+        final boolean proxyHostSet = validationContext.getProperty(PROXY_HOST).isSet();
+        final boolean proxyHostPortSet = validationContext.getProperty(PROXY_HOST_PORT).isSet();
+        if (proxyHostSet ^ proxyHostPortSet ) {
```

Comment:
The [Java operators](https://docs.oracle.com/javase/tutorial/java/nutsandbolts/opsummary.html) consider the XOR operator ("^") to be a bitwise operator not a conditional logic operator.  Even if it works correctly on this line, I think it would be more clear and less likely to be misinterpreted if the logical expressions were spelled out explicitly.
`        if ((proxyHostSet && !proxyHostPortSet) || (!proxyHostSet && proxyHostPortSet)) {`